File: app/src/main/java/com/example/sqlite/room/UserEntity.java
package com.example.sqlite.room;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.ColumnInfo;
import androidx.annotation.NonNull;

@Entity(tableName = "users")
public class UserEntity {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "user_id")
    public long userId;

    @NonNull
    @ColumnInfo(name = "email")
    public String email;

    @NonNull
    @ColumnInfo(name = "password_hash")
    public String passwordHash;

    @ColumnInfo(name = "created_at")
    public String createdAt;
}

File: app/src/main/java/com/example/sqlite/room/UserEntity.kt
package com.example.sqlite.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "users")
data class UserEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "user_id")
    var userId: Long = 0,

    @ColumnInfo(name = "email")
    var email: String = "",

    @ColumnInfo(name = "password_hash")
    var passwordHash: String = "",

    @ColumnInfo(name = "created_at")
    var createdAt: String? = null
)
